src/System/Hatrace.hs

```diff
@@ -2922,8 +2964,8 @@ instance ToJSON PTRACE_EVENT where
 
 
 -- | The terminology in here is oriented on `man 2 ptrace`.
-data TraceEvent
-  = SyscallStop SyscallStopType
+data TraceEvent a
+  = SyscallStop (SyscallStopType a)
```

Comment:
Probably good to rename `a` to `details` or `stopDetails` or `syscallStopDetails` to indicate meaning.

Same for `TraceState`'s `a` below.